src/Hazelcast.Net/DistributedObjects/IHDictionary.Getting.cs

```diff
@@ -21,98 +21,83 @@ namespace Hazelcast.DistributedObjects
-
-        /// <summary>
-        /// Gets all values.
-        /// </summary>
-        /// <returns>All values.</returns>
-        Task<IReadOnlyList<TValue>> GetValuesAsync();
-
+        Task<IReadOnlyCollection<TKey>> GetKeysAsync(IPredicate predicate);
+        
         /// <summary>
+        /// Queries the dictionary based on the specified predicate and returns a readonly collection of the values of matching entries.
         /// Gets values for entries matching a predicate.
         /// </summary>
-        /// <param name="predicate">An optional predicate to filter the entries.</param>
-        /// <returns>All values.</returns>
+        /// <param name="predicate">A predicate to filter the entries.</param>
+        /// <returns>readonly collection of the values of matching entries.</returns>
         /// <remarks>
+        /// <para>
+        /// Specified predicate runs on all members in parallel.
+        /// </para>
+        /// <para>
+        /// The returned collection is <b>NOT</b> backed by this dictionary,
+        /// so changes to the dictionary are <b>NOT</b> reflected in the collection, and vice-versa.
+        /// </para>
         /// <para>The <paramref name="predicate"/> must be serializable via Hazelcast serialization,
         /// and have a counterpart on the server.</para>
         /// </remarks>
-        Task<IReadOnlyList<TValue>> GetValuesAsync(IPredicate predicate);
-
-        /// <summary>
-        /// Gets the number of entries.
-        /// </summary>
-        /// <returns>The total number of entries in the map.</returns>
-        Task<int> CountAsync();
+        Task<IReadOnlyCollection<TValue>> GetValuesAsync(IPredicate predicate);
 
         /// <summary>
-        /// Determines whether this map is empty.
+        /// Queries the dictionary based on the specified predicate and returns a readonly dictionary of the matching entries.
         /// </summary>
-        /// <returns><c>true</c> if the map does not contain entries; otherwise <c>false</c>.</returns>
-        Task<bool> IsEmptyAsync();
+        /// <param name="predicate">A predicate to filter the entries with.</param>
+        /// <returns>readonly dictionary of the matching entries.</returns>
+        /// <remarks>
+        /// <para>
+        /// Specified predicate runs on all members in parallel.
+        /// </para>
+        /// <para>
+        /// The returned readonly dictionary is <b>NOT</b> backed by the hazelcast dictionary,
+        /// so changes to the returned readonly dictionary are <b>NOT</b> reflected in the <see cref="IHDictionary{TKey,TValue}"/>,
+        /// and vice-versa.
+        /// </para>
+        /// <para>
+        /// The <paramref name="predicate"/> must be serializable via Hazelcast serialization,
+        /// and have a counterpart on the server.</para>
+        /// </remarks>
+        Task<IReadOnlyDictionary<TKey, TValue>> GetEntriesAsync(IPredicate predicate);
```

Comment:
OK I am still not sure about this. `Get` gets a single thing, OK. `GetAll` gets everything, OK. But we are using `GetAll` with a collection of keys, and `GetEntries` with a predicate, that does not feel consistent. Thinking here...
* `Get` returns _one_ thing
* `GetAll` returns _everything_, unfiltered
* `GetMany` returns _many_ things, either from a list of keys, a predicate, whatever
* `GetStats` returns entry stats (and we rename `IHDictionaryEntry` into `IHDictionaryStats`

What do you think?